processor/batchprocessor/batch_processor.go

```diff
@@ -141,6 +143,16 @@ func (bp *batchTraceProcessor) startProcessingCycle() {
 	for {
 		select {
 		case td := <-bp.newTraceItem:
+			if bp.enforceBatchSize {
+				if bp.batchTraces.getSpanCount()+uint32(td.SpanCount()) > bp.sendBatchSize {
+					tds := splitTraces(int(bp.sendBatchSize-bp.batchTraces.getSpanCount()), int(bp.sendBatchSize), td)
+					for i := 1; i < len(tds); i++ {
+						go bp.ConsumeTraces(context.Background(), tds[i])
```

Comment:
It should go through `bp.sendItems` otherwise we lose self-reporting metrics about number of processed spans if I don't miss anything